tests/unit/stores/IndexStore.spec.js

```diff
@@ -0,0 +1,108 @@
+/**
+ * This source file is part of the Swift.org open source project
+ *
+ * Copyright (c) 2024 Apple Inc. and the Swift project authors
+ * Licensed under Apache License v2.0 with Runtime Library Exception
+ *
+ * See https://swift.org/LICENSE.txt for license information
+ * See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import IndexStore from 'docc-render/stores/IndexStore';
+import Language from 'docc-render/constants/Language';
+
+const flatChildren = [
+  {
+    title: 'item 1',
+  },
+  {
+    title: 'item 2',
+  },
+];
+
+const references = {
+  foo: { identifier: 'foo' },
+  bar: { identifier: 'bar' },
+};
+
+const apiChanges = {
+  interfaceLanguages: {
+    [Language.swift.key.url]: [],
+  },
+};
+
+const includedArchiveIdentifiers = ['foo', 'bar'];
+
+const technologyProps = {
+  [Language.swift.key.url]: {
+    technology: 'title',
+    technologyPath: 'path',
+    isTechnologyBeta: false,
+  },
+};
+
+describe('IndexStore', () => {
+  const defaultState = {
+    flatChildren: {},
+    references: {},
+    apiChanges: null,
+    includedArchiveIdentifiers: [],
+    errorFetching: false,
+    technologyProps: {},
+  };
+
+  beforeEach(() => {
+    localStorage.clear();
+    IndexStore.reset();
+  });
+
+  it('has a default state', () => {
+    expect(IndexStore.state).toEqual(defaultState);
+  });
+
+  describe('reset', () => {
+    it('restores the default state', () => {
+      IndexStore.state.flatChildren = flatChildren;
+      IndexStore.state.references = references;
+      IndexStore.state.apiChanges = apiChanges;
+      IndexStore.state.includedArchiveIdentifiers = includedArchiveIdentifiers;
+      IndexStore.state.errorFetching = true;
+      IndexStore.state.technologyProps = technologyProps;
+
+      expect(IndexStore.state).not.toEqual(defaultState);
+      IndexStore.reset();
+      // assert all the state is reset
+      expect(IndexStore.state).toEqual(defaultState);
+    });
+  });
+
+  it('sets `flatChildren`', () => {
+    IndexStore.setFlatChildren(flatChildren);
+    expect(IndexStore.state.flatChildren).toBe(flatChildren);
```

Comment:
I think you should use `toEqual` instead of `toBe` for any non-primitive JS types, like arrays of objects, so that it's checking for deep equality instead of reference equality.